docs/blog/2019-02-20-introducing-use-static-query/index.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Introducing useStaticQuery
+date: 2019-02-20
+author: Sidhartha Chatterjee
+tags:
+  - hooks
+  - static query
+---
+
+React Hooks are _cool_. Besides simplifying your code and removing the need for a lot of boilerplate associated with classes in JavaScript (looking at you, `this`), they enable some serious shareability. They also make it possible to use state in functional components.
+
+You can probably tell that we've been super excited about [React Hooks][hooks-intro]. So when they finally landed in React 16.8, we figured it was time to give our very own `StaticQuery` component the hook treatment!
+
+### Say hello to useStaticQuery
+
+`useStaticQuery` is a hook that takes a GraphQL query and returns your data. That's it.
+
+**No more Render Props necessary to use a Static Query**
+
+This simplifies accessing data in your components and also keeps your component tree shallow.
+
+Let's quickly check out a simple example. Here's a simple Header component, first using `StaticQuery` and then with `useStaticQuery`.
+
+### Before
+
+```jsx
+import React from "react"
+import { StaticQuery, graphql } from "gatsby"
+
+const Header = () => {
+  return (
+    <StaticQuery
+      query={graphql`
+        query {
+          site {
+            siteMetadata {
+              title
+            }
+          }
+        }
+      `}
+      render={data => (
+        <header>
+          <h1>{data.site.siteMetadata.title}</h1>
+        </header>
+      )}
+    />
+  )
+}
+
+export default Header
+```
+
+### After
+
+```jsx
+import React from "react"
+import { useStaticQuery, graphql } from "gatsby"
+
+const Header = () => {
+  const { site } = useStaticQuery(
```

Comment:
Would it be overkill to highlight the changes in these examples? Like lines 33, 42 and 44 in the "before" example, and 61, 73 and 75 in the after example (at least in the version I'm looking at).

Reply:
Definitely not overkill and I see the value in this! Done.